application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksSystem.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    boolean didUserBookmarkChannel(long userID, long channelID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(userID), BOOKMARKS.CHANNEL_ID.eq(channelID))
+            .fetch()
+            .isNotEmpty());
+    }
+
+    void addBookmark(long authorID, long channelID, @Nullable String note) {
+        database.write(context -> context.newRecord(BOOKMARKS)
+            .setAuthorId(authorID)
+            .setChannelId(channelID)
+            .setCreatedAt(Instant.now())
+            .setNote(note)
+            .setScheduledRemovalAt(null)
+            .insert());
+    }
+
+    List<BookmarksRecord> getUsersBookmarks(long authorID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .orderBy(BOOKMARKS.CREATED_AT.desc())
+            .fetch());
+    }
+
+    void removeBookmarks(long authorID, List<Long> channelIDs) {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID), BOOKMARKS.CHANNEL_ID.in(channelIDs))
+            .execute());
+    }
+
+    int getTotalBookmarkCount() {
+        return database
+            .read(context -> context.selectCount().from(BOOKMARKS).fetchOne(0, int.class));
+    }
+
+    int getUserBookmarkCount(long authorID) {
+        return database.read(context -> context.selectCount()
+            .from(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .fetchOne(0, int.class));
+    }
+
+    void scheduleUsersBookmarksRemoval(long authorID) {
+        Instant scheduledRemovalAt = Instant.now().plus(LEAVE_BOOKMARKS_REMOVAL_DELAY);
+
+        database.write(context -> context.update(BOOKMARKS)
+            .set(BOOKMARKS.SCHEDULED_REMOVAL_AT, scheduledRemovalAt)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .execute());
+    }
+
+    void cancelUsersBookmarksRemoval(long authorID) {
+        database.write(context -> context.update(BOOKMARKS)
+            .setNull(BOOKMARKS.SCHEDULED_REMOVAL_AT)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .execute());
+    }
+
+    void cleanRemovalScheduledBookmarks() {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.SCHEDULED_REMOVAL_AT.isNotNull(),
+                    BOOKMARKS.SCHEDULED_REMOVAL_AT.lessThan(Instant.now()))
+            .execute());
+    }
+
+    static MessageEmbed createSuccessEmbed(String content) {
+        return new EmbedBuilder().setDescription(content).setColor(COLOR_SUCCESS).build();
+    }
+
+    static MessageEmbed createWarningEmbed(String content) {
+        return new EmbedBuilder().setDescription(content).setColor(COLOR_WARNING).build();
+    }
+
+    static MessageEmbed createFailureEmbed(String content) {
+        return new EmbedBuilder().setDescription(content).setColor(COLOR_FAILURE).build();
+    }
```

Comment:
use a private helper that has the color as argument, which ur 3 methods here can use then.

also, u can make the colors private if u expect people to use these helper methods here

Reply:
So i will just go back to having the helper method i had before but still keep the 3 different color methods

Reply:
not sure what u had before. im okay if u keep those 3 methods, i just dont like their code duplication. so maybe let them forward to a private helper

Reply:
The method before had a second argument for color. Taz however requested this to be split into seperate methods. I do agree that always passing the color when generating the embeds looks bad but creating 3 methods just for colors also is not great. Maybe you can come up with a better solution

Reply:
I would not call it best of both worlds. Rather the lesser evil xD